### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by gemmen29**
### In this PR I added the following:
- Create a `classroom` class.
- Creating a `classroom has many students` relationships.
- Creating a `student belongs to a classroom` relationship.
- Create a `book` class.
- Create a `rental` class.
- Creating the `many-to-many` relationship between `person` and `book` using the intermediate class `rental`.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
```
Comment (SajjadAhmad14) [inline classroom.rb:10]: Please set the classroom for the student as well to make sure `has-many / belongs-to` association between `Classroom` and `Student`. You may read [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationshipsl) article for more details about the `has-many and belongs-to` association.

File: student.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require_relative './person'
 
 class Student < Person
+  attr_accessor :classroom
+
```
Comment (SajjadAhmad14) [inline student.rb:5]: Please add a method that serves as the 'belongs-to' side (a student belongs to a classroom). You may learn more about the association [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships).

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rentals(rental)
```
Comment (SajjadAhmad14) [inline book.rb]: Please create a new rental object of class Rental and pass `date`, `person`, and `book` itself as a parameter to achieve the `has-many` association.

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
```
Comment (SajjadAhmad14) [inline rental.rb:6]: Please add the rental itself to the @book to achieve the 'belongs-to' association. You may learn more about both ways association from [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships).

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    @person = person
```
Comment (SajjadAhmad14) [inline rental.rb:8]: Please add the rental itself to the @person to achieve 'belongs-to' association.

File: person.rb [hunk 6]
```diff
@@ -27,4 +28,8 @@ def can_use_services?
   def validate_name
     @name = @corrector.correct_name(@name)
   end
+
+  def add_rentals(rental)
+    @rentals << rental
```
Comment (SajjadAhmad14) [inline person.rb:35]: Please create new rental object of class Rental and pass `date`, `book`, and `person` itself as a parameter to achieve the `has-many` association. You may learn more about both ways association from [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships).

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi George, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: rental.rb (hunk 4 above)
Comment (gemmen29) [reply rental.rb:6]: Done.

File: rental.rb (hunk 5 above)
Comment (gemmen29) [reply rental.rb:8]: Done.

File: classroom.rb (hunk 1 above)
Comment (gemmen29) [reply classroom.rb:10]: Done.

File: student.rb (hunk 2 above)
Comment (gemmen29) [reply student.rb:5]: Done.

File: person.rb (hunk 6 above)
Comment (gemmen29) [reply person.rb:35]: Done.

File: book.rb (hunk 3 above)
Comment (gemmen29) [reply book.rb]: Done.

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @gemmen29, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**